### PR TITLE
✨ EUDEMO blanket

### PR DESCRIPTION
**Pull request by je-cook**
## Description

Recreate EUDEMO blanket, TODO: test

## Interface Changes

If you've had to update an interface or introduce a new interface as part of your change then let us know here.

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `flake8` and `black .`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+    def segment_blanket(self, silhouette: BluemiraFace, offset: float = 0.1):
+        """
+        Cut the blanket silhouette into segment silhouettes. Simple vertical cut for now.
+        """
+        bb = silhouette.bounding_box
+        x_mid = 0.5 * (bb.x_min + bb.x_max)
+        delta = 0.5 * self.params.c_rm.value
+
+        x = np.zeros(4) + x_mid
```
Comment (hsaunders1904) [inline examples/design/EU-DEMO/EUDEMO_builders/blanket.py:200]: Does `np.full(4, x_mid)` do the same thing here?

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    n_bb_inboard: Parameter[int]
+    n_bb_outboard: Parameter[int]
+    c_rm: Parameter[float]
+
+
+@parameter_frame
+class BlanketDesignerParams:
+    """
+    Blanket designer parameters
+    """
```
Comment (hsaunders1904) [inline examples/design/EU-DEMO/EUDEMO_builders/blanket.py:66]: Do we need this class if there are no parameters?

**Review by hsaunders1904 — APPROVED** [review]
A couple of nitpicks, but otherwise looks good. 

Could probably do with some tests - although I appreciate it can be hard to find things to assert about the geometry. Maybe we could assert we get the right number of segments (potentially, this could be a method on the `Blanket(ComponentManager)`? And, or, we could assert some bounds in the xyz dimensions. Happy with whatever you think best, but we should at least check that it runs.

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                segment.name, BluemiraFace(slice_shape(segment.shape, xy_plane)[0])
+            )
+            single_slice.plot_options.face_options["color"] = BLUE_PALETTE[self.BB][i]
+            slices.append(single_slice)
+
+        return circular_pattern_component(
+            Component(self.BB, children=slices), self.params.n_TF.value
+        )
+
+    def build_xyz(self, segments: List[PhysicalComponent], degree: float = 360.0):
```
[the remark this answers is not in the document]
Comment (je-cook) [reply examples/design/EU-DEMO/EUDEMO_builders/blanket.py:152]: I was waiting to see once I've added tests. Its very much situation dependent as #1352 tries to show

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py (hunk 2 above)
Comment (je-cook) [reply examples/design/EU-DEMO/EUDEMO_builders/blanket.py:66]: Forgot to commit that thanks :)

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (je-cook) [reply examples/design/EU-DEMO/EUDEMO_builders/blanket.py:123]: Seems to work in this case

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        x_mid = 0.5 * (bb.x_min + bb.x_max)
+        delta = 0.5 * self.params.c_rm.value
+
+        x = np.full(4, x_mid)
+        x[
+            (0, 3),
+        ] -= delta
+        x[
+            (1, 2),
+        ] += delta
```
Comment (hsaunders1904) [inline examples/design/EU-DEMO/EUDEMO_builders/blanket.py:179]: Good job, black 😆. Would removing the trailing comma improve the formatting here?

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+
+class Blanket(ComponentManager):
+    """
+    Wrapper around a Blanket component tree.
+    """
+
+
+@parameter_frame
+class BlanketBuilderParams:
```
Comment (hsaunders1904) [inline examples/design/EU-DEMO/EUDEMO_builders/blanket.py:50]: Did we decide to use `@dataclass` + inheritance when declaring these frames? This syntax doesn't play well with the typing system

**Review by hsaunders1904 — APPROVED** [review]

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py (hunk 5 above)
Comment (je-cook) [reply examples/design/EU-DEMO/EUDEMO_builders/blanket.py:50]: yep missed one :)

File: examples/design/EU-DEMO/EUDEMO_builders/blanket.py (hunk 4 above)
Comment (je-cook) [reply examples/design/EU-DEMO/EUDEMO_builders/blanket.py:179]: The indexing then moans that I'm indexing too axes. I will have a play

**Review by hsaunders1904 — APPROVED** [review]

**Pull request merged**